src/ui/PrintableUri/PrintableUri.ts

```diff
@@ -39,142 +40,79 @@ export class PrintableUri extends Component {
       }
-
       let parent = <Element>parents.item(i);
       let token = this.buildHtmlToken(parent.getAttribute('name'), parent.getAttribute('uri'));
       tokens.push(token);
-      element.appendChild(token);
-    }
 
-    if (tokens.length > 1) {
-      let ellipsis: HTMLElement = this.buildEllipsis();
-      element.insertBefore(ellipsis, seperators[0]);
-      let ellipsisSeperator: HTMLElement = this.buildSeperator();
-      element.insertBefore(ellipsisSeperator, ellipsis);
-
-      let contentWidth = 0;
-      let tokensWidth: number[] = [];
-      for (let i = 0; i < tokens.length; i++) {
-        tokensWidth[i] = tokens[i].offsetWidth;
-        contentWidth += tokensWidth[i];
-      }
-      let seperatorWidth = seperators[0].offsetWidth;
-      let ellipsisWidth = ellipsis.offsetWidth;
-      let availableWidth = element.offsetWidth;
-
-      if (availableWidth <= contentWidth) {
-        contentWidth += ellipsisWidth + seperatorWidth;
-        let hidden: HTMLElement[] = [];
-        let i = 1;
-        while (i < tokens.length && availableWidth <= contentWidth) {
-          element.removeChild(tokens[i]);
-          element.removeChild(seperators[i - 1]);
-          if (i > 1) {
-            hidden.push(seperators[i - 1]);
-          }
-          hidden.push(tokens[i]);
-          contentWidth -= tokensWidth[i] + seperatorWidth;
-          i++;
-        }
-        ellipsis.onclick = () => {
-          for (let i = 0; i < hidden.length; i++) {
-            element.insertBefore(hidden[i], ellipsis);
-          }
-          element.removeChild(ellipsis);
-        };
-      } else {
-        element.removeChild(ellipsis);
-        element.removeChild(ellipsisSeperator);
-      }
+      element.appendChild(token);
     }
   }
 
   public renderUri(element: HTMLElement, result?: IQueryResult) {
-    this.uri = result.clickUri;
-    let stringAndHoles: StringAndHoles;
-    if (result.printableUri.indexOf('\\') == -1) {
-      stringAndHoles = StringAndHoles.shortenUri(result.printableUri, $$(element).width() / 7);
+    let parentsXml = Utils.getFieldValue(result, 'parents');
+    if (parentsXml) {
+      this.renderParentsXml(element, parentsXml);
     } else {
-      stringAndHoles = StringAndHoles.shortenPath(result.printableUri, $$(element).width() / 7);
+      if (!this.options.titleTemplate) {
+        this.uri = result.clickUri;
+        let stringAndHoles: StringAndHoles;
+        if (result.printableUri.indexOf('\\') == -1) {
+          stringAndHoles = StringAndHoles.shortenUri(result.printableUri, $$(element).width() / 7);
+        } else {
+          stringAndHoles = StringAndHoles.shortenPath(result.printableUri, $$(element).width() / 7);
+        }
+        this.shortenedUri = HighlightUtils.highlightString(stringAndHoles.value, result.printableUriHighlights, stringAndHoles.holes, 'coveo-highlight');
+        let link = $$('div');
+        link.setAttribute('title', result.printableUri);
```

Comment:
this can all be replaced with 
   
   $$('div', {
      className : 'coveo-printable-uri-part',
      href : result.clickUri,
      title : result.printableUri
   }